git_machete/cmd.py

```diff
@@ -1570,15 +1517,246 @@ def flush_caches(self) -> None:
+        rmt: Optional[str] = self.__git.inferred_remote_for_fetching_of_branch(b)
+        self.__print_new_line(False)
+        if rmt:
+            self.handle_untracked_branch(rmt, b)
+        elif len(rems) == 1:
+            self.handle_untracked_branch(rems[0], b)
+        elif "origin" in rems:
+            self.handle_untracked_branch("origin", b)
+        else:
+            # We know that there is at least 1 remote, otherwise 's' would be 'NO_REMOTES'
+            print(fmt(f"Branch `{bold(b)}` is untracked and there's no `{bold('origin')}` repository."))
+            self.__pick_remote(b)
+
+    def __handle_ahead_state(self, cb: str, remote: str, abort_on_quit: bool = True) -> None:
+        self.__print_new_line(False)
+        ans = self.ask_if(
+            f"Push {bold(cb)} to {bold(remote)}?" + pretty_choices('y', 'N', 'q', 'yq'),
+            f"Pushing {bold(cb)} to {bold(remote)}...",
+            override_answer=None if self.__cli_opts.opt_push_tracked else "N"
+        )
+        if ans in ('y', 'yes', 'yq'):
+            self.__git.push(remote, cb)
+            if ans == 'yq':
+                if abort_on_quit:
+                    raise StopTraversal
+            self.flush_caches()
+        elif ans in ('q', 'quit'):
+            if abort_on_quit:
+                raise StopTraversal
+
+    def __handle_diverged_and_older_state(self, b: str) -> None:
+        self.__print_new_line(False)
+        rb = self.__git.strict_counterpart_for_fetching_of_branch(b)
+        ans = self.ask_if(
+            f"Branch {bold(b)} diverged from (and has older commits than) its remote counterpart {bold(rb)}.\nReset branch {bold(b)} to the commit pointed by {bold(rb)}?" + pretty_choices(
+                'y', 'N', 'q', 'yq'),
+            f"Branch {bold(b)} diverged from (and has older commits than) its remote counterpart {bold(rb)}.\nResetting branch {bold(b)} to the commit pointed by {bold(rb)}...")
+        if ans in ('y', 'yes', 'yq'):
+            self.__git.reset_keep(rb)
+            if ans == 'yq':
+                raise StopTraversal
+            self.flush_caches()
+        elif ans in ('q', 'quit'):
+            raise StopTraversal
+
+    def __handle_behind_state(self, b: str, remote: str) -> None:
+        rb = self.__git.strict_counterpart_for_fetching_of_branch(b)
+        ans = self.ask_if(f"Branch {bold(b)} is behind its remote counterpart {bold(rb)}.\n"
+                          f"Pull {bold(b)} (fast-forward only) from {bold(remote)}?" + pretty_choices('y', 'N', 'q', 'yq'),
+                          f"Branch {bold(b)} is behind its remote counterpart {bold(rb)}.\nPulling {bold(b)} (fast-forward only) from {bold(remote)}...")
+        if ans in ('y', 'yes', 'yq'):
+            self.__git.pull_ff_only(remote, rb)
+            if ans == 'yq':
+                raise StopTraversal
+            self.flush_caches()
+            print("")
+        elif ans in ('n', 'N', 'q', 'quit'):
```

Comment:
`ask_if` returns a lowercase string (see the impl), pls remove upper case letters from the similar checks everywhere